pkg/gofr/gofr.go

```diff
@@ -177,6 +178,29 @@ func (a *App) Run() {
 	wg.Wait()
 }
 
+// Shutdown stops the service(s) and close the application.
+func (a *App) Shutdown(ctx context.Context) (err error) {
+
+	if a.httpServer != nil {
+		err = errors.Join(err, a.httpServer.Shutdown(ctx))
+
+	}
+
+	if a.grpcServer != nil {
+		err = errors.Join(err, a.grpcServer.Shutdown(ctx))
+	}
+
+	if a.metricServer != nil {
+		err = errors.Join(err, a.metricServer.Shutdown(ctx))
+	}
+
+	err = errors.Join(err, a.container.Close())
+
+	a.container.Logger.Infof("Application shutdown complete: %v", err)
```

Comment:
So the log with be always present 

But the message will be strange with the trailing ": ' when there is no error, right?

And if an error occurs, it should be reported as a warning/error, no?